### PR TITLE
[SQL] Add tests for TINYINT

### PR DESCRIPTION
**Pull request by abhizer**
Fixes: #1145

Is this a user-visible change (yes/no): no

The tests are hand written taking inspiration from Postgres Integer tests.

### CODE REVIEW

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/TinyintTests.java [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+    @Test @Ignore("Integer wrapping: https://github.com/feldera/feldera/issues/1186")
+    public void testSelectOverflow() {
+        String error_message = "TINYINT out of range";
+        this.qf("SELECT i.f1, i.f1 * 2::TINYINT AS x FROM INT_TBL i", error_message);
+        this.qf( "SELECT i.f1, i.f1 + '2'::TINYINT AS x FROM INT_TBL i", error_message);
+        this.qf("SELECT i.f1, i.f1 - '2'::TINYINT AS x FROM INT_TBL i", error_message);
+    }
+
+    // This passes for the Calcite version but fails for the run time version
+    @Test @Ignore("Modulo edge case integer overflow: https://github.com/feldera/feldera/issues/1195")
```
Comment (mihaibudiu) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/TinyintTests.java:329]: There is no reason this should fail. There must be a bug in our rust library. The result is 0

**Review by mihaibudiu — APPROVED** [review]

**Comment by mihaibudiu** [comment]
I approved but there is one bug to fix. I don’t know why the ci failed.

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/TinyintTests.java (hunk 1 above)
Comment (abhizer) [reply sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/TinyintTests.java:329]: Yes, this is the same issue as with other integer types. 

https://github.com/feldera/feldera/pull/1196 fixes this.

Comment (mihaibudiu) [reply sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/TinyintTests.java:329]: Can we merge 1196?

Comment (abhizer) [reply sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/TinyintTests.java:329]: Yes, the merge conflicts have been resolved.
